pkg/generators/git.go

```diff
@@ -48,8 +48,8 @@ func (g *GitGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.Applic
 		return nil, EmptyAppSetGeneratorError
 	}
 
-	if appSetGenerator.Git == nil {
-		return nil, EmptyAppSetGeneratorError
+	if appSetGenerator.ApplicationSetTerminalGenerator == nil || appSetGenerator.ApplicationSetTerminalGenerator.Git == nil {
+		return nil, nil
```

Comment:
These checks are inconsistent. Some return `EmptyAppSetGeneratorError` and others return `nil`. I've standardized on `nil` for no particular reason.

These lines should never be hit. As far as I can tell they're just safety checks because the types technically allow this condition.

Reply:
Actually I would do the opposite as returning nil in the second parameter could lead in a possible `nil pointer error` for consumers of this function eventually. I would standardize and return `EmptyAppSetGeneratorError` everywhere when this condition is met.

Reply:
That seems completely reasonable. Will do.